juspay/hyperswitch
# feat(business_profile): add profile id in affected tables and modify api contract

## Type of Change



## Description

This PR will add `profile_id` in affected tables and also make necessary changes to the api contract to start accepting `profile_id`. The notable changes are
- Accept `default_profile` in the merchant account update to support having default business profile for a merchant.
- Merchant account create now creates a default business profile.
- Accept `profile_id` in the merchant connector create to associate a connector account with a business profile. 
- Accept `profile_id` in payments request.

### Additional Changes



migrations/2023-08-16-112847_add_profile_id_in_affected_tables/up.sql

## Motivation and Context



## How did you test it?

- Create a merchant account, the `default_profile` should be populated.
<img width="394" alt="Screenshot 2023-08-21 at 10 40 54 AM" src="https://github.com/juspay/hyperswitch/assets/48803246/7aac40d0-3073-4b99-8313-d49e47779ae5">

- Create a connector account, the default profile should be used.
<img width="883" alt="Screenshot 2023-08-21 at 10 42 23 AM" src="https://github.com/juspay/hyperswitch/assets/48803246/f1fcc879-1ef9-4181-b455-1b1f7c913489">

- Create a payment without business profile.  Default profile should be used.
<img width="696" alt="Screenshot 2023-08-21 at 12 09 26 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/83f16edd-a5f8-4986-9bf3-47e227f75f8a">

- Create a payment by passing business profile in the request.
<img width="736" alt="Screenshot 2023-08-21 at 2 37 13 PM" src="https://github.com/juspay/hyperswitch/assets/48803246/f1eae187-8ad2-4823-9877-e1d4c58ca0e4">

## Checklist
